### PR TITLE
BUG: df.sparse.to_coo() raising on duplicate colnames

### LINKED ISSUE #29564: Error getting COO of sparse matrix built from concatenation of series with same name
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

s = pd.Series([1, 2, 3], name="x")
df = pd.concat([s, s**2], axis=1)
df = df.astype(pd.SparseDtype("float", fill_value=0))
df.sparse.to_coo()
```
#### Problem description

The above code throws `IndexError: 0`.

This is a problem either because it is an unsupported case but the error message is *cryptic* or because it is a supported case.

#### Expected Output

No error is thrown or an exception explaining that columns with identical names are not supported is thrown.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.candidate.1
python-bits      : 64
OS               : Linux
OS-release       : 5.3.0-19-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.3
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.1.0
Cython           : 0.29.14
pytest           : None
hypothesis       : None
sphinx           : 2.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.10.3
IPython          : 7.9.0
pandas_datareader: None
bs4              : 4.8.0
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.1
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.3.2
sqlalchemy       : 1.3.11
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by mzeitlin11**
- [x] closes #29564
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/sparse/accessor.py [hunk 1]
```diff
@@ -339,12 +339,11 @@ def to_coo(self):
             dtype = dtype.subtype
 
         cols, rows, data = [], [], []
-        for col, name in enumerate(self._parent):
-            s = self._parent[name]
-            row = s.array.sp_index.to_int_index().indices
+        for col, (_, ser) in enumerate(self._parent.iteritems()):
```
Comment (jreback) [inline pandas/core/arrays/sparse/accessor.py:342]: can you use itertuples here? can you run the asv's to see if any perf implications

Comment (mzeitlin11) [reply pandas/core/arrays/sparse/accessor.py:342]: This is iterating over columns, so don't think itertuples would work here. Yep, will run asvs later

**Comment by mzeitlin11** [comment]
Added an ASV since frame case wasn't previously covered, no change

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**